### ISSUE TITLE
Failed building wheel for llama-cpp-python

### ISSUE THREAD

**Issue by smileyjay66** [imartinez/privateGPT#786]
I really don't know what to say, I've done everything correctly, I've cloned privateGTP to my MacBook but every time I run "pip3 install -r requirements.txt" I am met with this 

An error occurred while building with CMake.
        Command:
          /private/var/folders/jq/5pqnzgqd6jxfw6zbg1pt51vh0000gn/T/pip-build-env-v2huuux_/overlay/lib/python3.10/site-packages/cmake/data/bin/cmake --build . --target install --config Release --
        Install target:
          install
        Source directory:
          /private/var/folders/jq/5pqnzgqd6jxfw6zbg1pt51vh0000gn/T/pip-install-ksu_o96w/llama-cpp-python_d48452bdf4f34387a677dd78ec6b4e04
        Working directory:
          /private/var/folders/jq/5pqnzgqd6jxfw6zbg1pt51vh0000gn/T/pip-install-ksu_o96w/llama-cpp-python_d48452bdf4f34387a677dd78ec6b4e04/_skbuild/macosx-11.0-x86_64-3.10/cmake-build
      Please check the install target is valid and see CMake's output for more information.
      
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for llama-cpp-python
Failed to build llama-cpp-python
ERROR: Could not build wheels for llama-cpp-python, which is required to install pyproject.toml-based projects

yes I have both cmake and its extensions downloaded, ive tried the alternate code for macOS systems and that doesn't work either, please anyone that can help would be greatly appreciated

**Comment by Crescentz** [comment]
THE same question

**Comment by samanemami** [comment]
Update your CMake on Mac

**Comment by BasiL331** [comment]
On one machine I got it to work by installing Microsoft Visual C++ compiler after following advice from PromtEngineer/localGPT/issues#78.

On another I work through similar issues, trying to install gpt4all from source with cmake.

- If you installed cmake manually, first make sure to add it to the PATH environment variable.
- After that I am still stuck with a command like `cmake -CMAKE_CXX_COMPILER=/path/to/compiler/bin/g++` ignoring the parameter and falling back on the default compiler (which is MSVC and is not recommended [here](MSVC)). To solve that I currently expect that I must look at how the MinGW64 compiler is installed. There are different options to download it, I chose a portable zip. file. Maybe some installation needs to be run after unpacking at destination and before paths such as in the command above work.

**Comment by BasiL331** [comment]
On one machine I got it to work by installing Microsoft Visual C++ compiler after following advice from PromtEngineer/localGPT/#78.

On another I work through similar issues, trying to install gpt4all from source with cmake. I can share that experience:

First make sure to add CMake **and a compiler** you have next to it to the PATH environment variable.

After that I am still stuck with a command like cmake -CMAKE_CXX_COMPILER=/path/to/compiler/bin/g++ ignoring the parameter and falling back on the default compiler (which is MSVC and is not recommended [here](https://github.com/imartinez/privateGPT/issues/MSVC)). 
- It could be that I must look at how the MinGW64 compiler is installed. There are different options to download it, I chose a portable zip-file. Maybe some installation needs to be run (after unzipping) before paths such as in the command above work.
- Could also be that the file gpt4all-backend/CMakeLists.txt is incompatible with what I'm trying to do.
